apps/emqx_cluster_link/src/emqx_cluster_link_api.erl

```diff
@@ -7,86 +7,454 @@
+    with_link(Name, fun() -> handle_update(Name, Params0) end, not_found());
+'/cluster/links/link/:name'(delete, #{bindings := #{name := Name}}) ->
+    with_link(
+        Name,
+        fun() ->
+            case emqx_cluster_link_config:delete(Name) of
+                ok ->
+                    ?NO_CONTENT;
+                {error, Reason} ->
+                    Message = list_to_binary(io_lib:format("Delete link failed ~p", [Reason])),
+                    ?BAD_REQUEST(Message)
+            end
+        end,
+        not_found()
+    ).
+
+'/cluster/links/link/:name/metrics'(get, #{bindings := #{name := Name}}) ->
+    with_link(Name, fun() -> handle_metrics(Name) end, not_found()).
 
 %%--------------------------------------------------------------------
 %% Internal funcs
 %%--------------------------------------------------------------------
 
+handle_list() ->
+    Links = get_raw(),
+    NodeResults = get_all_link_status_cluster(),
+    NameToStatus = collect_all_status(NodeResults),
+    EmptyStatus = #{status => inconsistent, node_status => []},
+    Response =
+        lists:map(
+            fun(#{<<"name">> := Name} = Link) ->
+                Status = maps:get(Name, NameToStatus, EmptyStatus),
+                maps:merge(Link, Status)
+            end,
+            Links
+        ),
+    ?OK(Response).
+
+handle_create(Name, Params) ->
+    case emqx_cluster_link_config:create(Params) of
+        {ok, Link} ->
+            ?CREATED(add_status(Name, Link));
+        {error, Reason} ->
+            Message = list_to_binary(io_lib:format("Create link failed ~p", [Reason])),
+            ?BAD_REQUEST(Message)
+    end.
+
+handle_lookup(Name, Link) ->
+    ?OK(add_status(Name, Link)).
+
+handle_metrics(Name) ->
+    case emqx_cluster_link:get_metrics(Name) of
+        {error, BadResults} ->
+            ?SLOG(warning, #{
+                msg => "cluster_link_api_metrics_bad_erpc_results",
+                results => BadResults
+            }),
+            ?OK(#{metrics => #{}, node_metrics => []});
+        {ok, NodeResults} ->
+            NodeMetrics =
+                lists:map(
+                    fun({Node, Metrics}) ->
+                        format_metrics(Node, Metrics)
+                    end,
+                    NodeResults
+                ),
+            AggregatedMetrics = aggregate_metrics(NodeMetrics),
+            Response = #{metrics => AggregatedMetrics, node_metrics => NodeMetrics},
+            ?OK(Response)
+    end.
+
+aggregate_metrics(NodeMetrics) ->
+    ErrorLogger = fun(_) -> ok end,
+    lists:foldl(
+        fun(#{metrics := Metrics}, Acc) ->
+            emqx_utils_maps:best_effort_recursive_sum(Metrics, Acc, ErrorLogger)
+        end,
+        #{},
+        NodeMetrics
+    ).
+
+format_metrics(Node, Metrics) ->
+    Routes = emqx_utils_maps:deep_get([counters, ?route_metric], Metrics, 0),
+    #{
+        node => Node,
+        metrics => #{
```

Comment:
I think we could as well expose resource metrics of the message forwarding resource?

Reply:
:checkered_flag: